### ISSUE TITLE
[FEATURE REQUEST] Using "albumsortorder" tag to chronologically sort albums

### ISSUE THREAD

**Issue by HermeticAttic** [namidaco/namida#424]
TL;DR
I believe this is a very useful and "powerful" option for any library elements such as Songs, Artists, Albums, Genres etc.

And now I'll try to explain this idea with an example.
Let's say we have albums by one artist in chronological order:
01 - Steineiche (1998)
02 - Schattengang (1998)
03 - Die Festung (1998)
04 - Paysage d'Hiver (1999)

If we open artist page from Artists tab then we will see a list of this albums (horizontal block) in alphabetical order and songs sorted by year:
![Artist page](https://github.com/user-attachments/assets/ca888627-b458-4487-b39c-93a69b4c5dfd)

For song sorting I used the next settings:
![Sort settings](https://github.com/user-attachments/assets/0a48376b-b833-46ee-98bd-f9be22d88dcb)

The problem is that within the same year, the albums are sorted alphabetically, i.e.
Die Festung (1998)
Schattengang (1998)
Steineiche (1998)
which is different from the actual order (see above).

To solve similar problem in my music library on PC I add the full release date to the "albumsortorder" tag, for example:
1998.01.01 - Steineiche
1998.05.01 - Schattengang
1998.12.01 - Die Festung
1999.01.01 - Paysage d’Hiver
etc.
It looks like this:
![foobar2000](https://github.com/user-attachments/assets/f53cb25f-bbb3-4c0d-9bb2-8a8a6799da5e)
![iTunes](https://github.com/user-attachments/assets/42b67f8b-638d-40ff-9541-1255a85d028d)

And now the albums are displayed in chronological order, not of alphabetical:
![Library view](https://github.com/user-attachments/assets/19094fd9-f6a6-4be7-8d6f-eb154a815029)

So I think if Namida can use this tag for sorting (let's call it "Sort by Release Date" for example), it it will be very handy for library navigation and very useful for those who like to listen to albums in the order they were released. Oh, and apply this sorting to albums view in Artist page (that horizontal block in the first screenshot) too, please?

Thanks.

**Comment by MSOB7YY** [comment]
interesting, im quite curious of why not to use the full date in the year tag itself??

i believe "sort album", "sort artist" are custom tags by itunes and not widely used. tbh they seem quite useless to me unless i really want a custom sort

so how about u edit the year tag like this
19980101 - Steineiche
19980501 - Schattengang
19981201 - Die Festung
19990101 - Paysage d’Hiver

**Comment by HermeticAttic** [comment]
> so how about u edit the year tag like this
19980101 - Steineiche
19980501 - Schattengang
19981201 - Die Festung
19990101 - Paysage d’Hiver

Because it will completely destroy any grouping by year in any music player, not just in iTunes.
For examle foobar2000 with library view by year:
![Screenshot 2024-12-13 135526](https://github.com/user-attachments/assets/5d91bc32-fac3-47d9-8c67-ac52c484aa09)

Just imagine what it would look like if I write "Year" tag in YYYYMMDD-format.

Yes, maybe "albumsortorder" is [custom tags by itunes](https://exiftool.org/TagNames/ID3.html#v2_2), but that doesn't mean nothing else can use it. For another example (and again foobar2000) - [this](https://www.reddit.com/r/foobar2000/comments/198wj68/how_do_i_change_albums_sorting_order/) reddit post about usig this tag. And in Google I've found a lot of requests for developers to add this sorting to their music players - so which means a lot of people know about it.

**Comment by MSOB7YY** [comment]
> Just imagine what it would look like if I write "Year" tag in YYYYMMDD-format.

i thought most players take only the first 4 digits representing the year (can u confirm that?)

> Yes, maybe "albumsortorder" is [custom tags by itunes](https://exiftool.org/TagNames/ID3.html#v2_2), but that doesn't mean nothing else can use it

the point is the moment u start supporting custom tags, u are now guranteed to felixibly support *any* tag out there, which is not realistic.

altho album_sort_order seems kinda important, what about artist sort & genre sort etc? 


anyways i will try to add it soon, pls send 1 track for 2 different albums to test

**Comment by HermeticAttic** [comment]
> i thought most players take only the first 4 digits representing the year (can u confirm that?)

Of course I can't speak for everyone, and sometimes I've seen non-standard entries in "Year" tag, but personally I only use it for year in 4-digit format.

> altho album_sort_order seems kinda important, what about artist sort & genre sort etc?

I'm sorry, I'm not sure I understand the question correctly :(
I think if the player works with metadata, not files, they are all important.

> anyways i will try to add it soon, pls send 1 track for 2 different albums to test

Thank you for your interest in this. I'm not sure about better-way for sharing track for test, so let it be a [third-party service](https://www.mediafire.com/file/ryyquhvkrwlcdo8/for_test.zip/file). They're two tracks from two previously mentioned albums with the required tags.

**Comment by OmniSIM** [comment]
Excuse my question if you've already addressed it but is there any updates pertaining to this?

![Image](https://github.com/user-attachments/assets/85b9289a-00b3-4200-9d9d-e79852bd70e2)

I would also appreciate a feature to sort the horizontal information by years/release dates

**Comment by MSOB7YY** [comment]
@HermeticAttic sorry for being so late, this feature is now implemented in v5.8.2 https://github.com/namidaco/namida-snapshots/releases

@OmniSIM this is a separate thing from the original issue, u are asking for a way to sort horizontal albums in artist page, you should know that they follow the main sorting of the main albums page, if u sort by `title`, these will also be sorted by `title`.
sry it was requested few times before but there are no plans to separate sorts for these, they just follow the main sort

**Comment by HermeticAttic** [comment]
@MSOB7YY thanks you, all this time I didn't lose hope 👍 
I checked the latest beta, and I have one last (I think) question.

As you mentioned above, in main albums tab I use new sort `Album (sort by)` (1) and it works great. When I switch to artist page - this sorting is used to horizontal albums list too (2), that's cool.
But below on this screen, in tracks section, this new sorting is not applied (3) and is not available in the "Sort by" (4) menu at all. :(
Is it possible to fix this in future releases?

For the illustrations below, I used windows app, but it's same in android too.

![Image](https://github.com/user-attachments/assets/8b7015a9-6b68-4356-a91a-15e95d1dfa1c)

**Comment by MSOB7YY** [comment]
@HermeticAttic oh so u would like to sort even tracks by "album sort" tag? i haven't thought about that lmao
btw is "title sort" a thing or is it useless for sorting tracks?
